### DISCUSSION TITLE
Braindecode: how to add features to the BCNI2014001 dataset trials

### DISCUSSION THREAD

**Discussion by fportera** [braindecode/braindecode#831] [General]
Hello,
we are trying to concatenate some spectral Granger causality features to the dataset cited in the subject.
We have an x_train np array with shape (1296, 22231), and we have an y_train np array for the targets with shape (1296).
How can we create a training dataset?
Many thanks for the help you can provide.

**Comment by fportera** [comment]
We reshaped our arrays in order to call create_from_X_y(), but the resulting training was well below expectations.
Meaning that the test accuracy is lower with sgc concatenated features than without sgc features.

**Reply by fportera** [reply]
For the dataset described in the subject, considering 2 classes (left and right hand motion imagery), we reached a test accuracy of about 0.918. The net is a combination of CNN and Transformer, with an ensemble of 5 trials.

**Comment by fportera** [comment]
While, for the 4 classes task, with the same net, we reached a test accuracy of about 0.77 .

**Comment by bruAristimunha** [comment]
Hey!

Okay, you have your own way of evaluate the data and you're getting this result, then you convert the dataset using Braindecode and you get different results. 

That's strange, there is potentially some detail wrong, the question that we need solve is where. 

What we need now is a Minimum Viable Code that reproduces this effect. When I say Minimum, I really mean the minimum, 30 lines-ish maximum inside one an Colab.

Perhaps only by showing the difference in the PyTorch dataset instance.

I await your return :)
